### PR TITLE
Grant follower permissions

### PR DESCRIPTION
**Pull request by j-vent**
Load all of the incoming follow requests made to the user (social map value 0 or 2) into listview

Added a function to user controller to filter only requests (not users we are following)

Add functionality to the accept button in listview

Updated the database using controller to make changes to the current user's social map and the newly-following users' social map. I followed the accept case detailed in the [design doc](https://docs.google.com/document/d/1AvdjGUVFGz1nljk_wC7BOVH3f-RHT9iHm-OkQ3fQJ_k/edit)

I accept a friend request:
If friend == 0 -> None (Delete my friend in my social map and my friend successfully follows me)
Elif friend == 2 -> 1 (My friend successfully follows me and I continue to follow my friend)

NOTE: I have been testing this on 1 user only (the user who is getting these follow requests). The "other users" have been updated on firebase directly. Will do more thorough
testing once I have added the "deny" case.

## Screenshots

follower_1_incoming
This is the initial state of my social map. I am following june (value = 1) and there are 2 pending follow requests that I can accept (0 and 2).
follower_2_following
We can see here that 2 --> 1 (we are still following the user who requested to follow us) and chee's follow request disappeared from my social map
follower_3_otheruserfollowsusnow
We can see that justine's user ID has been added to chee's list as he is now following me (justine's value = 1)

follower_gif

## Checklist

- [ ] Automated tests
- [X] Screenshots included (if necessary)
- [X] Code is well commented

Closes #ISSUE_ID

### CODE REVIEW

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java [hunk 1]
```diff
@@ -24,29 +28,57 @@
...
-        // add a fake user to arrayList
-        UserModel fakeUser = new UserModel();
-        fakeUser.setFirstName("Fake");
-        fakeUser.setLastName("McFakerson");
-        fakeUser.setUsername("fk123");
-        followRequestDataList.add(fakeUser);
-
         followRequestAdapter = new FollowRequestCustomList(this, followRequestDataList);
         followRequestList.setAdapter(followRequestAdapter);
+        userController = new UserController();
```
Comment (cheetiantey) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java:58]: nit: Add a comment to this logic

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java [hunk 2]
```diff
@@ -24,29 +28,57 @@
...
+//                            followRequestUser.getFirstName() + " " + followRequestUser.getUID());
+                    followRequestDataList.add(followRequestUser);
+                    // update adapter that new users have been added to list
+                    followRequestAdapter.notifyDataSetChanged();
+                });
+                // followRequestList.addAll(hbEvtLst);
+
+            }
+        });
+       followRequestList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
```
Comment (cheetiantey) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java:73]: nit: Add a comment to this listener

**Review by cheetiantey — APPROVED** [review]
I left some nit comments, but other than that LGTM!

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java (hunk 1 above)
Comment (j-vent) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java:58]: user controller is already defined above with a comment!

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java (hunk 2 above)
Comment (j-vent) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java:73]: that listener was not relevant, will remove!

Comment (cheetiantey) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/FollowRequestActivity.java]: Got it!

**Review by cheetiantey — APPROVED** [review]

**Review by Docker-J — APPROVED** [review]

**Pull request merged**